### PR TITLE
update_build_tools_image.yml: Push build-tools image to Docker Hub

### PR DESCRIPTION
**Pull request by bayandin**
## Problem

- `docker.io/neondatabase/build-tools:pinned` image is frequently outdated on Docker Hub because there's no automated way to update it.
- `update_build_tools_image.yml` workflow contains legacy roll-back logic, which is not required anymore because it updates only a single image.

## Summary of changes
- Make `update_build_tools_image.yml` workflow push images to both ECR and Docker Hub
- Remove unneeded roll-back logic

```
$ docker manifest inspect neondatabase/build-tools:test2
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2079,
         "digest": "sha256:b2aab29ac81e1e2a3314fa4c0fda2bdde2180768dcd6f6e2d8e15a097f520123",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2079,
         "digest": "sha256:d3fbb151accd1621cf7b6e1f0909a9de67ac7b3fb99a5741ba54a7b9adb1498b",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      }
   ]
}
```

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Review by abhigets — APPROVED** [review]

File: .github/workflows/update_build_tools_image.yml [hunk 1]
```diff
@@ -20,111 +20,55 @@ defaults:
...
-          go install github.com/google/go-containerregistry/cmd/crane@a54d64203cffcbf94146e04069aae4a97f228ee2 # v0.16.1
-          go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login@adf1bafd791ae7d4ff098108b1e91f36a4da5404 # v0.7.1
+          mkdir -p .docker-custom
+          echo DOCKER_CONFIG=$(pwd)/.docker-custom >> $GITHUB_ENV
 
-      - name: Configure ECR login
-        run: |
-          mkdir /github/home/.docker/
-          echo "{\"credsStore\":\"ecr-login\"}" > /github/home/.docker/config.json
+      - uses: docker/login-action@v2
```
Comment (ololobus) [inline .github/workflows/update_build_tools_image.yml:43]: Did you test it? Last time I tried something like that there was a very nontransparent error, which boiled down to that you cannot login to both ECR and docker hub at the same time. At least I'm pretty sure I had this issue in GitHub Actions

**Review by ololobus — COMMENTED** [review]
Thanks

> docker image supports both linux/amd64 and linux/arm64 platforms

Do we need to do something additionally here? Last time I tried Docker image from ECR worked just fine on ARM

File: .github/workflows/update_build_tools_image.yml (hunk 1 above)
Comment (bayandin) [reply .github/workflows/update_build_tools_image.yml:43]: Yep, it works https://github.com/neondatabase/neon/actions/runs/7660677085/job/20878475165

The trick here is to use custom `DOCKER_CONFIG`. We have some content in default `~/.docker/config.json` which doesn't play well with `docker/login-action@v2`

**Comment by bayandin** [comment]
> > docker image supports both linux/amd64 and linux/arm64 platforms
> 
> Do we need to do something additionally here? Last time I tried Docker image from ECR worked just fine on ARM

By default, docker pulls the image for the current platform, this implementation copies only `linux/amd64`, we need to add handling for `linux/arm64` explicitly (there're tools like `skopeo` or `crane` that can handle multi-platform images, I will try them, but not sure if they pick up docker's authorisation)

**Comment by bayandin** [comment]
`crane` works nicely with multi-platform manifests:
```
$ docker manifest inspect neondatabase/build-tools:test2
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2079,
         "digest": "sha256:b2aab29ac81e1e2a3314fa4c0fda2bdde2180768dcd6f6e2d8e15a097f520123",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2079,
         "digest": "sha256:d3fbb151accd1621cf7b6e1f0909a9de67ac7b3fb99a5741ba54a7b9adb1498b",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      }
   ]
}
```

**Pull request merged**